core/math/quat.h

```diff
@@ -83,9 +84,9 @@ class Quat {
 
 	_FORCE_INLINE_ Vector3 xform(const Vector3 &v) const {
 
-		Quat q = *this * v;
-		q *= this->inverse();
-		return Vector3(q.x, q.y, q.z);
+		Vector3 u(x, y, z);
+		Vector3 uv = u.cross(v);
+		return v + ((uv * w) + u.cross(uv)) * ((real_t)2);
```

Comment:
Just to be sure, does this give the exact same result as the previous calculation method, just faster?
It would be nice to have a short explanatory comment as it's a bit opaque as is.

Reply:
None of the math code has that sort if explanation though.

It does the same thing when the quaternion is normalized. It's just a simplification using a few vector algebra identities and it's probably best to add that to rotation docs. This is what glm uses BTW.

Reply:
It's normal for Quat methods to assume that it's normalized BTW, just like Basis is assumed to be orthonormalize. Can add a debug only check though.

Reply:
But, I think we should add debug only checks for that in various places, including this function and inverse() and some others.

Reply:
Actually a Quat is supposed to be a unit quaternion, so I agree, it should be mentioned

Reply:
I just checked the document and the definition of Quat is totally wrong, I'll fix it in an update. A quaterion isn't a 4D vector or a 4D complex number, those are totally different thing. A quaternion is is a quaternion.

Reply:
To be more specific, Quat in game engines is one of elements Spin(3) corresponding to the desired SO(3) rotation, in quaternion representation.

In that regard, talking about the properties of quaternion algebra in Quat's definition is similar to talking about properties of matrices in the definition of Basis, which is a bit off-topic.